cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/AppLifecycleHttpHandler.java

```diff
@@ -609,13 +610,22 @@ private List<ApplicationId> decodeAndValidateBatchApplication(NamespaceId namesp
           throw new BadRequestException("Missing 'appId' in the request element.");
         }
         String appId = obj.get("appId").getAsString();
+        String version = null;
 
-        JsonElement version = obj.get("version");
-        if (version == null) {
-          result.add(validateApplicationId(namespaceId, appId));
+        JsonElement versionObj = obj.get("version");
```

Comment:
`/appdetail` returns the latest version if version is not specified in the request